src/mpid/ch4/shm/posix/posix_rma.h

```diff
@@ -207,25 +207,27 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_do_get_accumulate(const void *origin_ad
 
     if (MPIDIG_WIN(win, shm_allocated)) {
         MPIDI_POSIX_RMA_MUTEX_LOCK(posix_win->shm_mutex_ptr);
-        shm_locked = 1;
+    } else {
+        MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
     }
 
     mpi_errno = MPIR_Localcopy((char *) base + disp_unit * target_disp, target_count,
                                target_datatype, result_addr, result_count, result_datatype);
-    MPIR_ERR_CHECK(mpi_errno);
 
-    if (op != MPI_NO_OP) {
+    if (mpi_errno == MPI_SUCCESS && op != MPI_NO_OP) {
```

Comment:
nitpick: we could move the unlock block below the `fn_exit` label and avoid this error checking change.

Reply:
Then we'll need add another flag variable, e.g. `need_unlock`, to distinguish the case from earlier `goto fn_exit`. I think we'd prefer one less variable.

Reply:
I was thinking like an `fn_unlock` label which could unlock and goto `fn_exit`. Probably not worth over-thinking. It would be really great to re-think these labels if function logging is ever removed.

Reply:
@raffenet Did we resolve this comment?

Reply:
Yes, we can leave this as-is.